docs/developer-tools/ThrowHelper.md

```diff
@@ -0,0 +1,153 @@
+
+// ...with this
+ThrowHelper.ThrowInvalidOperationException("Some custom message from my library");
+```
+
+The `Guard` APIs provide an easy to use abstraction over many common checks that might need to be performed, so it is recommended to first try to see if there is a `Guard` API that can be used, and only fallback on using `ThrowHelper` if that's not the case. As mentioned before, this might be if your library needs to throw a specific exception type not present in the `Guard` APIs, or if you need full control on the exact arguments being included in the exception being thrown.
+
+> [!NOTE]
+> It is not necessary to understand all the technical details behind this, the general idea is simply that `throw new Exception(...)` statements can safely be replaced by `ThrowHelper.ThrowException(...)` calls to achieve better performance. If you're interested in learning more about this, the next paragraph will describe the actual differences in the generated code in more detail.
+
+## What makes the `ThrowHelper` pattern special
+
+To see why this is necessary, we'll need to have a peek at the code that the JIT compiler is actually producing in these situations. We'll use the code generated on .NET Core 3.1 x64 as a reference, but the same concept applies to other runtimes and architectures as well.
+
+Let's consider this simple type:
+
+```csharp
+public struct ExceptionTest
+{
+    private int number;
+    
+    public int GetValueIfNotZeroOrThrow()
+    {
+        if (number == 0)
+        {
+            throw new InvalidOperationException("The value must be != 0");
+        }
+        
+        return number;
+    }
+}
+```
+
+This type is only meant to showcase the use of `ThrowHelper` APIs. Here we're simply checking the value of the `number` field, and throwing if it's 0. Let's check the generated asm code:
+
+```x86asm
+ExceptionTest.GetValueIfNotZeroOrThrow()
+    L0000: push rsi
+    L0001: sub rsp, 0x20
+    L0005: mov eax, [rcx]   ; read number
+    L0007: test eax, eax    ; if (number != 0)
+    L0009: je short L0011   ; jump to faulting path (if == 0)
+    L000b: add rsp, 0x20
+    L000f: pop rsi
+    L0010: ret              ; return (number is already in eax)
+    L0011: mov rcx, 0x7ff95cbaca80  ; exception setup...
+    L001b: call 0x00007ff9bc6178f0
+    L0020: mov rsi, rax
+    L0023: mov ecx, 1
+    L0028: mov rdx, 0x7ff96590c080
+    L0032: call 0x00007ff9bc7403e0
+    L0037: mov rdx, rax
+    L003a: mov rcx, rsi
+    L003d: call System.InvalidOperationException..ctor(System.String)
+    L0042: mov rcx, rsi
+    L0045: call 0x00007ff9bc5db3a0
+    L004a: int3 ; finally throw the exception
+```
+
+We can see the code contains a lot of lines just dedicated to creating the exception being thrown. This can cause a large increase in the code size whenever we have many of these checks, because it will reduce the efficacy of the instruction cache, etc. In general, we'd like the size of our methods to be as small as possible.
```

Comment:
`efficiacy` is valid here, but i think `efficienc`y is just clearer

Reply:
Whoops, meant to write that indeed ahahah 😄
Done in https://github.com/MicrosoftDocs/WindowsCommunityToolkitDocs/pull/370/commits/4b1c607a14d6150626a27d0b7edcbfdc9ecee357, thanks!